util/design/gen-top-docs.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+r"""Generates top level documentation from an hjson configuration file."""
+
+import argparse
+import logging as log
+from pathlib import Path
+
+import hjson
+from lib import common
+from tabulate import tabulate
+
+
+def to_markdown(text):
+    """Adds prefix and sufix newlines to 'text'."""
+    doc_text = f"""
+{text}
+"""
+    return doc_text
+
+
+def generate_mmap_table(top_level):
+    """Generates top level memory map table."""
+    header = ["Name", "Type", "Byte Address"]
+    table = [header]
+    colalign = ("left", ) * len(header)
+
+    for module in top_level["module"]:
+        for j, (name, base) in enumerate(module["base_addrs"].items()):
+
+            base_address = f"{base} ({name})"
+            if name == "null":
+                base_address = f"{base} (regs)"
+
+            if j == 0:
+                row = [module["name"], module["type"], base_address]
```

Comment:
Should we print out the size of the region as well?

Reply:
Will do as part of a refactoring follow up.